### ISSUE TITLE
Feature suggestion: use ciso8601 in pd.to_datetime for great speed

### ISSUE THREAD

**Issue by MichaMucha** [pandas-dev/pandas#21666]
**tl;dr** [ciso8601](https://github.com/closeio/ciso8601) is a module that's over 450x faster than pandas.to_datetime for data in ISO8601 format

#### Problem description

pd.to_datetime is robust to various formats and works nice.

I was trying to convert a series of 3.5 million timestamps into python objects, and it was taking long enough for me to start looking for alternatives. Found **[this.](https://www.peterbe.com/plog/fastest-python-datetime-parser)**

Result:
<img width="726" alt="screen shot 2018-06-28 at 9 31 05 pm" src="https://user-images.githubusercontent.com/7082264/42060047-69304f1c-7b1d-11e8-8792-164ce926cdac.png">

I'm on the 2017 macbook pro, pandas was using one core.  
I tried on 100k observations as well - 12s pandas vs 43ms ciso  (280x)
Your mileage may vary, but it's safe to say the improvement is significant enough for bigger datasets.

Does it make sense to try and take advantage of this nice treat? If yes, how to go about it?

Thanks

**Comment by WillAyd** [comment]
Need a reproducible example to work off of to get on the same page. Also need to know your version information - please be sure to reference the contributing guide on the best way to open bug reports:

https://pandas.pydata.org/pandas-docs/stable/contributing.html#bug-reports-and-enhancement-requests

**Comment by MichaMucha** [comment]
Wow, you guys are great! So fast, and you're right Chris, the ` UTC` ruined everything

Noted for the future. Will post text

Will: I'm using pandas 0.23.1

Results on the full 3.5million
```python
%timeit pd.to_datetime(df.account_created_at.str.slice(stop=-4))
1.49 s ± 52.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

%timeit [ciso8601.parse_datetime(x) for x in df.account_created_at.str.slice(stop=-4)]
1.56 s ± 19.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

%timeit pd.to_datetime([ciso8601.parse_datetime(x) for x in df.account_created_at.str.slice(stop=-4)])
1.94 s ± 148 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Thanks again I guess you can close this

**Comment by WillAyd** [comment]
Thanks @chris-b1
